pulpcore/app/migrations/0060_data_migration_proxy_creds.py

```diff
@@ -0,0 +1,47 @@
+# Generated by Django 2.2.18 on 2021-03-03 10:59
+
+from urllib.parse import urlparse, urlunparse
+
+from django.db import migrations
+
+
+def move_proxy_creds_up(apps, schema_editor):
+    Remote = apps.get_model('core', 'Remote')
+    for remote in Remote.objects.all():
+        if remote.proxy_url and "@" in remote.proxy_url:
+            try:
+                parsed_url = urlparse(remote.proxy_url)
+                netloc = parsed_url.netloc
+                auth, url = netloc.rsplit("@", maxsplit=1)
+                remote.proxy_username, remote.proxy_password = auth.split(":", maxsplit=1)
+                remote.proxy_url = urlunparse(parsed_url._replace(netloc=url))
+                remote.save()
+            except Exception:
+                pass
+
+
+def move_proxy_creds_down(apps, schema_editor):
+    Remote = apps.get_model('core', 'Remote')
+    for remote in Remote.objects.all():
+        if remote.proxy_url and remote.proxy_username and remote.proxy_password:
+            try:
+                parsed_url = urlparse(remote.proxy_url)
+                url = parsed_url.netloc
+                netloc = remote.proxy_username + ":" + remote.proxy_password + "@" + url
+                remote.proxy_username = None
+                remote.proxy_password = None
+                remote.proxy_url = urlunparse(parsed_url._replace(netloc=netloc))
+                remote.save()
+            except Exception:
+                pass
```

Comment:
Python 3,4+ added `suppress` to avoid the use of `pass` in broader exceptions like this. https://docs.python.org/3.6/library/contextlib.html#contextlib.suppress

```suggestion
            with contextlib.suppress(Exception)
                parsed_url = urlparse(remote.proxy_url)
                url = parsed_url.netloc
                netloc = remote.proxy_username + ":" + remote.proxy_password + "@" + url
                remote.proxy_username = None
                remote.proxy_password = None
                remote.proxy_url = urlunparse(parsed_url._replace(netloc=netloc))
                remote.save()
```

Reply:
Interesting!